src/NServiceBus.Core/Serializers/MessageDeserializerResolver.cs

```diff
@@ -1,15 +1,24 @@
 ﻿namespace NServiceBus
 {
+    using System;
     using System.Collections.Generic;
-    using System.Linq;
     using Serialization;
 
     class MessageDeserializerResolver
     {
         public MessageDeserializerResolver(IMessageSerializer defaultSerializer, IEnumerable<IMessageSerializer> additionalDeserializers)
         {
             this.defaultSerializer = defaultSerializer;
-            serializersMap = additionalDeserializers.ToDictionary(key => key.ContentType, value => value);
+
+            foreach (var additionalDeserializer in additionalDeserializers)
+            {
+                if (serializersMap.ContainsKey(additionalDeserializer.ContentType))
+                {
+                    throw new Exception($"Multiple deserializers are registered for content-type '{additionalDeserializer.ContentType}'. Remove ambiguous deserializers.");
```

Comment:
@andreasohlund I think it is good as is. To come up with a proper message we would need to pass in the serializer definition. Just outputting the serializer type is not helpful. That type could be internal and even have a completely different name than the serializer definition (which is the public type the user uses)